### ISSUE TITLE
Bad request error when attempting to send push notification

### ISSUE THREAD

**Issue by iAmWillShepherd** [OneSignalDevelopers/onesignal-go-sample#2]
No matter which OneSigna key I use for app authentication, I get the following error. 

```shell
╭─iamwill@kronos ~/code/@onesignalDevelopers/onesignal-go-sample ‹main› 
╰─$ go run .
oAuthCtx: context.Background.WithValue(type onesignal.contextKey, val )
Error when calling `CreateNotification`: 400 Bad Request
Full HTTP response: &{400 Bad Request 400 HTTP/2.0 2 0 map[Access-Control-Allow-Headers:[SDK-Version] Access-Control-Allow-Origin:[*] Alt-Svc:[h3=":443"; ma=86400, h3-29=":443"; ma=86400] Cache-Control:[no-cache] Cf-Cache-Status:[DYNAMIC] Cf-Ray:[74d52062a9cee85b-DFW] Content-Type:[application/json; charset=utf-8] Date:[Mon, 19 Sep 2022 20:38:42 GMT] Referrer-Policy:[strict-origin-when-cross-origin] Server:[cloudflare] Status:[400 Bad Request] Strict-Transport-Security:[max-age=15552000; includeSubDomains] Vary:[Accept-Encoding] Via:[1.1 google] X-Content-Type-Options:[nosniff] X-Download-Options:[noopen] X-Envoy-Upstream-Service-Time:[8] X-Frame-Options:[SAMEORIGIN] X-Permitted-Cross-Domain-Policies:[none] X-Request-Id:[1b942caa-4ee2-4a42-a58e-b9cea9ec3c5f] X-Runtime:[0.006427] X-Xss-Protection:[1; mode=block]] {{"errors":["Please include a case-sensitive header of Authorization: Basic \u003cYOUR-REST-API-KEY-HERE\u003e or Bearer token=\"\u003cYOUR-REST-API-KEY-HERE\u003e\" with a valid REST API key."],"reference":["https://documentation.onesignal.com/docs/accounts-and-keys#section-keys-ids"]}} -1 [] false true map[] 0x14000230100 0x140001720b0}
```

###### REST_API_KEY

```go
var osAuthCtx = context.WithValue(
	context.Background(),
	onesignal.AppAuth,
	os.Getenv("REST_API_KEY"),
)
```

###### USER_AUTH_KEY

```go
notification := *onesignal.NewNotification(os.Getenv("APP_ID"))
```

**Comment by kesheshyan** [comment]
Hey Will,

I've managed to get it working replacing env variables with the local ones, also added a couple required fields to the notification object:

```go
package lib

import (
	"context"
	"fmt"
	"os"
	"github.com/OneSignal/onesignal-go-api"
)

var appId = "replace_with_your_app_id"
var restApiKey = "replace_with_your_api_key"

var configuration = onesignal.NewConfiguration()
var apiClient = onesignal.NewAPIClient(configuration)
var osAuthCtx = context.WithValue(
	context.Background(),
	onesignal.AppAuth,
	restApiKey,
)

func PushNotification() {
	fmt.Fprintf(os.Stdout, "oAuthCtx: %v\n", osAuthCtx)
	
	notification := *onesignal.NewNotification(appId)
	notification.SetIncludedSegments([]string{"Subscribed Users"})
	notification.SetIsIos(false)
	testMessage := "Go Test Notification"
	stringMap := onesignal.StringMap{En: &testMessage}
	notification.Contents = *onesignal.NewNullableStringMap(&stringMap)

	resp, r, err := apiClient.DefaultApi.CreateNotification(osAuthCtx).Notification(notification).Execute()

	if err != nil {
		fmt.Fprintf(os.Stderr, "Error when calling `CreateNotification`: %v\n", err)
		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
		return
	}

	fmt.Fprintf(os.Stdout, "Response from `CreateNotification`: %v\n", resp)
	fmt.Fprintf(os.Stdout, "Notification ID: %s", resp.GetId())
}
```

**Comment by iAmWillShepherd** [comment]
The issue turned out to be environment variables not being in scope when variables are assigned. This was hard to diagnose due to the environment variables being around during runtime. @kesheshyan and I sorted this out when we noticed hard-coding the values worked after confirming that the environment variables contained the same value at runtime. I had a hunch that this was a program lifecycle issue. I confirmed this by moving variable assignments inside the function based on my hypothesis that environment variables weren't defined until runtime.
